aurora-mysql-plugin/src/test/java/co/cask/aurora/mysql/AuroraMysqlSinkTestRun.java

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * Test for ETL using databases.
+ */
+public class AuroraMysqlSinkTestRun extends AuroraMysqlPluginTestBase {
+
+  @Test
+  public void testDBSink() throws Exception {
+    String inputDatasetName = "input-dbsinktest";
+
+    ETLPlugin sourceConfig = MockSource.getPlugin(inputDatasetName);
+    ETLPlugin sinkConfig = new ETLPlugin(
+      AuroraMysqlConstants.PLUGIN_NAME,
+      BatchSink.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .putAll(BASE_PROPS)
+        .put(AbstractDBSink.DBSinkConfig.TABLE_NAME, "MY_DEST_TABLE")
+        .put(Constants.Reference.REFERENCE_NAME, "DBTest")
+        .build(),
+      null);
+
+    deployETL(sourceConfig, sinkConfig, DATAPIPELINE_ARTIFACT, "testDBSink");
+    createInputData(inputDatasetName);
+
+
+    try (Connection conn = createConnection();
+         Statement stmt = conn.createStatement();
+         ResultSet resultSet = stmt.executeQuery("SELECT * FROM my_table")) {
+      Set<String> users = new HashSet<>();
+      Assert.assertTrue(resultSet.next());
+      users.add(resultSet.getString("NAME"));
+      Assert.assertEquals(new Date(CURRENT_TS).toString(), resultSet.getDate("DATE_COL").toString());
+      Assert.assertEquals(new Time(CURRENT_TS).toString(), resultSet.getTime("TIME_COL").toString());
+      Assert.assertEquals(new Timestamp(CURRENT_TS),
+                          resultSet.getTimestamp("TIMESTAMP_COL"));
+      Assert.assertTrue(resultSet.next());
+      users.add(resultSet.getString("NAME"));
+      Assert.assertEquals(ImmutableSet.of("user1", "user2"), users);
+
+    }
+  }
+
+  private void createInputData(String inputDatasetName) throws Exception {
+    // add some data to the input table
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    Schema schema = Schema.recordOf(
+      "dbRecord",
+      Schema.Field.of("ID", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("NAME", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("SCORE", Schema.of(Schema.Type.FLOAT)),
+      Schema.Field.of("GRADUATED", Schema.of(Schema.Type.BOOLEAN)),
+      Schema.Field.of("TINY", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("SMALL", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("BIG", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("FLOAT_COL", Schema.of(Schema.Type.FLOAT)),
+      Schema.Field.of("REAL_COL", Schema.of(Schema.Type.FLOAT)),
+      Schema.Field.of("NUMERIC_COL", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("DECIMAL_COL", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("BIT_COL", Schema.of(Schema.Type.BOOLEAN)),
+      Schema.Field.of("DATE_COL", Schema.of(Schema.LogicalType.DATE)),
+      Schema.Field.of("TIME_COL", Schema.of(Schema.LogicalType.TIME_MICROS)),
+      Schema.Field.of("TIMESTAMP_COL", Schema.of(Schema.LogicalType.TIMESTAMP_MICROS)),
+      Schema.Field.of("BINARY_COL", Schema.of(Schema.Type.BYTES)),
+      Schema.Field.of("BLOB_COL", Schema.of(Schema.Type.BYTES))
```

Comment:
just making sure, these are all possible field types?

Reply:
This covers primitive and logical types.